### PR TITLE
[terra-button] Remove the default margin in Safari.

### PR DESCRIPTION
**Pull request by nramamurth**
### Summary
Resolves https://github.com/cerner/terra-clinical/issues/471

### CODE REVIEW

File: packages/terra-button/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Unreleased
 ------------------
 ### Changed
 * Update react/react-dom peer-dependency to latest stable minor release
+* Reset default margin in Safari. [#471](https://github.com/cerner/terra-clinical/issues/471)
```
Comment (ShettyAkarsh) [inline packages/terra-button/CHANGELOG.md]: No need to provide the issue link here.

Comment (emilyrohrbough) [reply packages/terra-button/CHANGELOG.md]: Lets move this under the unreleased heading under the section `### Fixed` to align with our [changelog conventions](https://keepachangelog.com/en/1.0.0/).  When we release we will update with the date and release version

File: packages/terra-button/src/Button.module.scss [hunk 2]
```diff
@@ -32,6 +32,7 @@
     font-family: inherit;
     font-size: var(--terra-button-font-size, 1rem);
     line-height: var(--terra-button-line-height, 1.429);
+    margin: 0; /* Reset the margin in Safari. */
```
Comment (yuderekyu) [inline packages/terra-button/src/Button.module.scss]: nit: want to add more context to the comment?
```suggestion
    margin: 0; /* Reset margin for Safari. Safari defaults to 2px */
```

Comment (nramamurth) [reply packages/terra-button/src/Button.module.scss]: Done - https://github.com/cerner/terra-core/pull/2316/commits/8b4ce473766ef3d3bcc2e3a7f8fc71df2ed25756

File: packages/terra-button/CHANGELOG.md (hunk 1 above)
Comment (nramamurth) [reply packages/terra-button/CHANGELOG.md]: Done - https://github.com/cerner/terra-core/pull/2316/commits/8b4ce473766ef3d3bcc2e3a7f8fc71df2ed25756.

**Pull request merged**